includes/addon-akismet.php

```diff
@@ -0,0 +1,185 @@
+<?php
+
+/**
+ * Scan doc content using the Akismet spam-filtering service.
+ *
+ * @since 2.1.0
+ */
+class BP_Docs_Akismet {
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 2.1.0
+	 */
+	public function __construct() {
+	}
+
+	/**
+	 * Add action and filter hooks to add Akismet functionality.
+	 *
+	 * @since 2.1.0
+	 */
+	public function add_hooks() {
+		// Check docs against the Akismet service.
+		add_filter( 'bp_docs_args_before_save', array( $this, 'check_for_spam'), 10, 3 );
+	}
+
+	/**
+	 * Check if the activity item is spam or ham.
+	 *
+	 * @since 2.1.0
+	 *
+	 * @see http://akismet.com/development/api/
+	 * @todo Auto-delete old spam?
+	 *
+	 * @param array  $r    The parameters to be used in wp_insert_post().
+	 * @param object $this The BP_Docs_Query object.
+	 * @param array  $args The passed and filtered parameters for the doc
+	 *                     about to be saved.
+	 */
+	public function check_for_spam( $save_args, $bdq_object, $passed_args ) {
+		// Build data package for Akismet.
+		$akismet_package = $this->build_akismet_data_package( $save_args, $passed_args );
+
+		// Check with Akismet to see if this is spam.
+		$akismet_response = $this->send_akismet_request( $akismet_package, 'check', 'spam' );
+
+		// Spam.
+		if ( 'true' == $akismet_response['bp_docs_as_result'] ) {
```

Comment:
It really sends back the string `'true'`?

Reply:
Ha ha ha, no, thankfully. 💯

Reply:
OK, well then I take it back:
```
akismet_response: array(2) {
  [0]=>
  object(Requests_Utility_CaseInsensitiveDictionary)#1403 (1) {
    ["data":protected]=>
    array(5) {
      ["server"]=>
      string(5) "nginx"
      ["date"]=>
      string(29) "Fri, 22 Dec 2017 15:58:16 GMT"
      ["content-type"]=>
      string(25) "text/plain; charset=utf-8"
      ["content-length"]=>
      string(1) "4"
      ["x-akismet-guid"]=>
      string(32) "long-string-xxxxx"
    }
  }
  [1]=>
  string(4) "true"
}
```
I feel a little better, anyway.

Reply:
I had a feeling they probably did, but I wanted to be sure in case you were getting fooled by the loose `==` :-D